src/orchestrator.rs

```diff
@@ -710,4 +732,62 @@ mod tests {
         let s = "哈囉世界";
         assert_eq!(index_codepoints(s, 3, 4), "界");
     }
+
+    // Test for TGIS generation with default parameter
+    #[tokio::test]
+    async fn test_tgis_generate_with_default_params() {
+        // Initialize a mock object from `TgisClient`
+        let mut mock_client = TgisClient::faux();
+
+        let sample_text = String::from("sample text");
+        let text_gen_model_id = String::from("test-llm-id-1");
+
+        let generation_response = GenerationResponse {
+            text: String::from("sample response worked"),
+            stop_reason: StopReason::EosToken.into(),
+            stop_sequence: String::from("\n"),
+            generated_token_count: 3,
+            seed: 7,
+            ..Default::default()
+        };
+
+        let client_generation_response = BatchedGenerationResponse {
+            responses: [generation_response].to_vec(),
+        };
+
+        let expected_generate_req_args: BatchedGenerationRequest = BatchedGenerationRequest {
```

Comment:
nit: drop explicit type

```suggestion
        let expected_generate_req_args = BatchedGenerationRequest {
```